benchmarks/tests.py

```diff
@@ -425,3 +425,45 @@ class TestJobTestCase(TestCase):
     def test_metadata_is_empty_by_default(self):
         job = TestJob()
         self.assertEqual({}, job.metadata)
+
+    def test_invalid_json(self):
+        job = TestJob(definition='//')
+        job.__extract_metadata__()
+        self.assertEqual({}, job.metadata)
+
+    def test_extract_metadata_from_empty_definition(self):
+        job = TestJob(definition=None)
+        job.__extract_metadata__() # just not crashing is good enough here
+
+    def test_extract_metadata_from_json_definition(self):
+        job = TestJob(definition='{ "metadata" : { "foo": "bar", "baz": "qux" } }')
+        job.__extract_metadata__()
+
+        self.assertEqual(job.metadata['foo'], 'bar')
+        self.assertEqual(job.metadata['baz'], 'qux')
+
+    def test_extract_metadata_from_nested_metadata(self):
+        job = TestJob(definition='{ "something": { "name": "foobar", "metadata" : { "foo": "bar", "baz": "qux" } } }')
+        job.__extract_metadata__()
+
+        self.assertEqual(job.metadata['foo'], 'bar')
+        self.assertEqual(job.metadata['baz'], 'qux')
+
+    def test_extract_metadata_from_lava_job(self):
```

Comment:
should we add test for extracting metadata from more than one action in the job?